speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp

```diff
@@ -190,6 +205,80 @@ static void SetUserId(void*)
+            break;
+        case TASK_SPEED_CONTROL:
+            GfParmSetStr(readParam, RESEARCHMENU_SPEED_CONTROL, GFMNU_ATTR_CHECKED, GFMNU_VAL_YES);
+            break;
+    }
+
+    // Save indicator settings to xml file
+    const char* audioSetting = m_indicators.Audio ? GFMNU_VAL_YES : GFMNU_VAL_NO;
+    GfParmSetStr(readParam, RESEARCHMENU_INDICATOR_AUDITORY, GFMNU_ATTR_CHECKED, audioSetting);
+    const char* iconSetting = m_indicators.Icon ? GFMNU_VAL_YES : GFMNU_VAL_NO;
+    GfParmSetStr(readParam, RESEARCHMENU_INDICATOR_VISUAL, GFMNU_ATTR_CHECKED, iconSetting);
+    const char* textSetting = m_indicators.Text ? GFMNU_VAL_YES : GFMNU_VAL_NO;
+    GfParmSetStr(readParam, RESEARCHMENU_INDICATOR_TEXT, GFMNU_ATTR_CHECKED, textSetting);
+
+    // Save intervention type settings to xml file
+    GfParmSetStr(readParam, RESEARCHMENU_TYPE_NO_SIGNALS, GFMNU_ATTR_CHECKED, GFMNU_VAL_NO);
+    GfParmSetStr(readParam, RESEARCHMENU_TYPE_ONLY_SIGNALS, GFMNU_ATTR_CHECKED, GFMNU_VAL_NO);
+    GfParmSetStr(readParam, RESEARCHMENU_TYPE_SHARED_CONTROL, GFMNU_ATTR_CHECKED, GFMNU_VAL_NO);
+    GfParmSetStr(readParam, RESEARCHMENU_TYPE_COMPLETE_TAKEOVER, GFMNU_ATTR_CHECKED, GFMNU_VAL_NO);
+    switch (m_interventionType)
+    {
+        case INTERVENTION_TYPE_NO_SIGNALS:
+            GfParmSetStr(readParam, RESEARCHMENU_TYPE_NO_SIGNALS, GFMNU_ATTR_CHECKED, GFMNU_VAL_YES);
+            break;
+        case INTERVENTION_TYPE_ONLY_SIGNALS:
+            GfParmSetStr(readParam, RESEARCHMENU_TYPE_ONLY_SIGNALS, GFMNU_ATTR_CHECKED, GFMNU_VAL_YES);
+            break;
+        case INTERVENTION_TYPE_SHARED_CONTROL:
+            GfParmSetStr(readParam, RESEARCHMENU_TYPE_SHARED_CONTROL, GFMNU_ATTR_CHECKED, GFMNU_VAL_YES);
+            break;
+        case INTERVENTION_TYPE_COMPLETE_TAKEOVER:
+            GfParmSetStr(readParam, RESEARCHMENU_TYPE_COMPLETE_TAKEOVER, GFMNU_ATTR_CHECKED, GFMNU_VAL_YES);
+            break;
+    }
+
+    // Save participant control settings to xml file
+    const char* controlGas = m_pControl.ControlGas ? GFMNU_VAL_YES : GFMNU_VAL_NO;
+    GfParmSetStr(readParam, RESEARCHMENU_CONTROL_GAS, GFMNU_ATTR_CHECKED, controlGas);
+    const char* interventionToggle = m_pControl.ControlInterventionToggle ? GFMNU_VAL_YES : GFMNU_VAL_NO;
+    GfParmSetStr(readParam, RESEARCHMENU_CONTROL_INTERVENTION_TOGGLE, GFMNU_ATTR_CHECKED, interventionToggle);
+    const char* controlSteering = m_pControl.ControlSteering ? GFMNU_VAL_YES : GFMNU_VAL_NO;
+    GfParmSetStr(readParam, RESEARCHMENU_CONTROL_STEERING, GFMNU_ATTR_CHECKED, controlSteering);
+    const char* forceFeedback = m_pControl.ForceFeedback ? GFMNU_VAL_YES : GFMNU_VAL_NO;
+    GfParmSetStr(readParam, RESEARCHMENU_FORCE_FEEDBACK, GFMNU_ATTR_CHECKED, forceFeedback);
+
+    // Save max time to xml file
+    char buf2[32];
+    sprintf(buf2, "%d", m_maxTime);
+    GfParmSetStr(readParam, RESEARCHMENU_MAX_TIME_EDIT, GFMNU_ATTR_DEFAULT_VALUE, buf2);
```

Comment:
buf(1) doesn't exist anymore, so buf2 doesn't make sense
```suggestion
    char buf[32];
    sprintf(buf, "%d", m_maxTime);
    GfParmSetStr(readParam, RESEARCHMENU_MAX_TIME_EDIT, GFMNU_ATTR_DEFAULT_VALUE, buf);
```